kpi/models/object_permission.py

```diff
@@ -443,45 +443,23 @@ def _get_effective_perms(
-                content_type.pk, codename__startswith='change_')
-
-            for change_perm_pk, change_perm_codename in change_permissions:
-                share_permission_codename = re.sub(
-                    '^change_', 'share_', change_perm_codename, 1)
-                if (codename is not None and
-                        share_permission_codename != codename
-                ):
-                    # If the caller specified `codename`, skip anything that
-                    # doesn't match exactly. Necessary because `Asset` has
-                    # `*_submissions` in addition to `*_asset`
-                    continue
-                share_perm_pk, _ = self.__get_permissions_for_content_type(
-                    content_type.pk,
-                    codename=share_permission_codename)[0]
-                for user_id, permission_id in effective_perms_copy:
-                    if permission_id == change_perm_pk:
-                        effective_perms.add((user_id, share_perm_pk))
-        # The owner has the delete_ permission
-        if self.owner is not None and (
-                user is None or user.pk == self.owner.pk) and (
-                codename is None or codename.startswith('delete_')
-        ):
-            delete_permissions = self.__get_permissions_for_content_type(
-                content_type.pk, codename__startswith='delete_')
-            for delete_perm_pk, delete_perm_codename in delete_permissions:
-                if (codename is not None and
-                        delete_perm_codename != codename
-                ):
-                    # If the caller specified `codename`, skip anything that
-                    # doesn't match exactly. Necessary because `Asset` has
-                    # `delete_submissions` in addition to `delete_asset`
-                    continue
-                effective_perms.add((self.owner.pk, delete_perm_pk))
+        # The owner has the share_ and delete_ permission
+        for codename_prefix in ('share_', 'delete_'):
+            if self.owner is not None and (
+                    user is None or user.pk == self.owner.pk) and (
+                    codename is None or codename.startswith(codename_prefix)
+            ):
+                matching_permissions = self.__get_permissions_for_content_type(
+                    content_type.pk, codename__startswith=codename_prefix)
+                for perm_pk, perm_codename in matching_permissions:
+                    if (codename is not None and
+                            perm_codename != codename
+                    ):
+                        # If the caller specified `codename`, skip anything that
+                        # doesn't match exactly. Necessary because `Asset` has
+                        # `delete_submissions` in addition to `delete_asset`
+                        continue
+                    effective_perms.add((self.owner.pk, perm_pk))
```

Comment:
I was wondering why the part of the code was removed until I looked above and saw
> # TODO: remove this flag; support for it has been removed from
> # ObjectPermissionMixin

I guess we will implement `share_asset` as an assignable permission one day?

Reply:
Indeed!